RELEASE_NOTES.txt

```diff
@@ -1,6 +1,120 @@
+* Added the ability to define a custom path for the snapshot operation.
+* Added the ability to intercept calls to service methods.
+* Consistency check now able to fix counters after the consistency fix.
+* Consistency check now support cacheGroup as a param as well.
+* Fixed 'Invalid cross-device link error' when extending WAL segments size.
+* Fixed ClassCastException that appears during deployment manager stop in Java versions 11.0.16+ and 17.0.4+.
+* Fixed Integer value is out of bounds when the dataregion maxSize is large enough.
+* Fixed an issue that caused a node is trying to resolve its hostname if IP is configured with IGNITE_LOCAL_HOST.
+* Fixed an issue that caused a node resolves obsolete addresses from the previously restarted and killed nodes.
+* Fixed an issue that caused inconsistent keys after deletion during rebalance.
+* Fixed an issue that could lead to a block of the management pool in case of using clearSync.
+* Fixed an issue that could lead to unexpected partition map exchange on client nodes.
+* Fixed an issue where the snapshot error was not propagated to the initiating node.
+* Fixed an issue with potential node failure during cancelling of compute jobs with cache operations.
+* Fixed configured event listeners resource injection order.
+* Fixed durable background task start on BLT node joined to the active cluster.
+* Fixed expiration doesn't apply for cache entries replicated with CDC.
+* Fixed incorrect params of GridCacheManager#stop method.
+* Fixed overflowing async operation permits maximum for explicit transaction with single write entry.
+* Fixed partition update counter tracking on primary after the node restart.
+* Fixed potential deadlock in transactions recovery on node failure.
+* Fixed propagation of a service call exception stacktrace to the client side.
+* Fixed race condition in GridNioServer.
+* Fixed remote partitions files move on snapshot restore operation.
+* Fixed run ignitevisorcmd in JDK 17 environment.
+* Fixed server-side message parsing for big messages from thin clients.
+* Fixed tasks to sent snapshot files.
+* Fixed the "io.datastorage.StorageSize" metric in case of multiple persistence dataregions.
+* Fixed the command `idle_verify` of the control.sh utility; the command prints and compares full partition counter state.
+* Fixed write empty binary object.
+* Fixed wrong checkpoint locks count calculation after lock timeout (wrong calculation leads to node failure in some cases).
+* Improved REST commands: skip authorization for  'Probe' and 'Version' commands.
+* Improved `--consistency repair` command of the control.sh utility; Read Repair now support arrays and collections as values.
+* Improved `--consistency repair` command of the control.sh utility; Read Repair now support binary keys.
+* Improved `--consistency repair` command of the control.sh utility; atomic caches now repairable by Read Repair.
```

Comment:
57. Improved `--consistency repair` command of the control.sh utility; atomic caches can now be repaired by Read Repair.